File: SozlukJava/app/src/main/java/com/example/sozlukjava/view/GirisFragment.java
package com.example.sozlukjava.view;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.sozlukjava.R;
import com.example.sozlukjava.databinding.FragmentGirisBinding;
import com.example.sozlukjava.model.SimpleResponse;
import com.example.sozlukjava.util.SessionManager;
import com.example.sozlukjava.viewmodel.GirisViewModel;

// GirisFragment.java
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.fragment.NavHostFragment;

public class GirisFragment extends Fragment {

    private FragmentGirisBinding binding;
    private GirisViewModel viewModel;
    private SessionManager session;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = FragmentGirisBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        session = new SessionManager(requireContext());

        // Oturum açık mı?
        if (session.isLoggedIn()) {
            NavHostFragment.findNavController(this)
                    .navigate(R.id.action_girisFragment_to_anaSayfaFragment);
            return;
        }

        viewModel = new ViewModelProvider(this).get(GirisViewModel.class);

        binding.btnGiris.setOnClickListener(v -> {
            String username = binding.etUsername.getText().toString();
            String password = binding.etPassword.getText().toString();
            viewModel.login(username, password);
        });

        viewModel.getLoginResult().observe(getViewLifecycleOwner(), new Observer<SimpleResponse>() {
            @Override
            public void onChanged(SimpleResponse it) {
                if (it == null) return;

                if (it.isSuccess()) {
                    int uid = (it.getUser_id() != null) ? it.getUser_id() : -1;
                    session.saveUserSession(uid, binding.etUsername.getText().toString());
                    NavHostFragment.findNavController(GirisFragment.this)
                            .navigate(R.id.action_girisFragment_to_anaSayfaFragment);
                } else {
                    String msg = (it.getMessage() != null) ? it.getMessage() : "Giriş başarısız";
                    Toast.makeText(requireContext(), msg, Toast.LENGTH_SHORT).show();
                }
            }
        });

        binding.btnKayitSayfasinaGit.setOnClickListener(v ->
                NavHostFragment.findNavController(GirisFragment.this)
                        .navigate(R.id.action_girisFragment_to_kayitFragment)
        );
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: SozlukKotlin/app/src/main/java/com/example/sozlukkotlin/view/GirisFragment.kt
package com.example.sozlukkotlin.view

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.fragment.findNavController
import com.example.sozlukkotlin.R
import com.example.sozlukkotlin.databinding.FragmentGirisBinding
import com.example.sozlukkotlin.util.SessionManager
import com.example.sozlukkotlin.viewmodel.GirisViewModel


class GirisFragment : Fragment() {
    private var _binding: FragmentGirisBinding? = null
    private val binding get() = _binding!!

    private lateinit var viewModel: GirisViewModel
    private lateinit var session: SessionManager

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        _binding = FragmentGirisBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        session = SessionManager(requireContext())

        // Oturum açık mı?
        if (session.isLoggedIn()) {
            findNavController().navigate(R.id.action_girisFragment_to_anaSayfaFragment)
            return
        }

        viewModel = ViewModelProvider(this)[GirisViewModel::class.java]

        binding.btnGiris.setOnClickListener {
            val username = binding.etUsername.text.toString()
            val password = binding.etPassword.text.toString()

            viewModel.login(username, password)
        }

        viewModel.loginResult.observe(viewLifecycleOwner) {
            if (it.success) {
                session.saveUserSession(it.user_id ?: -1, binding.etUsername.text.toString())
                findNavController().navigate(R.id.action_girisFragment_to_anaSayfaFragment)
            } else {
                Toast.makeText(requireContext(), it.message ?: "Giriş başarısız", Toast.LENGTH_SHORT).show()
            }
        }

        binding.btnKayitSayfasinaGit.setOnClickListener {
            findNavController().navigate(R.id.action_girisFragment_to_kayitFragment)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
